Sub pass ssa_opt_private_append

**Describe the bug**
I'm trying to compile my project (https://github.com/bragful/ephp) using OTP 26.2.5.1 and looks like it's experiencing an issue, first, I compiled and found the error about +0.00 but solving that, it's failing again with this error message:

```
./rebar3 escriptize
===> Verifying dependencies...
===> Analyzing applications...
===> Compiling ephp
Sub pass ssa_opt_private_append
===> Compiling src/ephp_parser.erl failed
src/ephp_parser.erl:none: internal error in pass beam_ssa_opt:
exception error: no function clause matching beam_ssa_private_append:patch_literal_term([],{hd,self},429)
  in function  beam_ssa_private_append:patch_opargs/6 (beam_ssa_private_append.erl, line 540)
  in call from beam_ssa_private_append:patch_opargs/3 (beam_ssa_private_append.erl, line 534)
  in call from beam_ssa_private_append:patch_appends_is/5 (beam_ssa_private_append.erl, line 488)
  in call from beam_ssa_private_append:patch_appends_f/5 (beam_ssa_private_append.erl, line 469)
  in call from beam_ssa_private_append:'-patch_appends/3-anonymous-2-'/3 (beam_ssa_private_append.erl, line 438)
  in call from maps:fold_1/4 (maps.erl, line 416)
  in call from beam_ssa_private_append:private_append/3 (beam_ssa_private_append.erl, line 72)

make: *** [Makefile:23: ephp] Error 1
```

**To Reproduce**
Check using these commands:
```
wget https://raw.githubusercontent.com/bragful/ephp/master/include/ephp.hrl
wget https://raw.githubusercontent.com/bragful/ephp/master/include/ephp_parser.hrl
wget https://raw.githubusercontent.com/bragful/ephp/master/src/ephp_parser.erl
erlc ephp_parser.erl
```
It should drop this error:
```
$ erlc ephp_parser.erl
Sub pass ssa_opt_private_append
ephp_parser.erl: internal error in pass beam_ssa_opt:
exception error: no function clause matching beam_ssa_private_append:patch_literal_term([],{hd,self},429)
  in function  beam_ssa_private_append:patch_opargs/6 (beam_ssa_private_append.erl, line 540)
  in call from beam_ssa_private_append:patch_opargs/3 (beam_ssa_private_append.erl, line 534)
  in call from beam_ssa_private_append:patch_appends_is/5 (beam_ssa_private_append.erl, line 488)
  in call from beam_ssa_private_append:patch_appends_f/5 (beam_ssa_private_append.erl, line 469)
  in call from beam_ssa_private_append:'-patch_appends/3-anonymous-2-'/3 (beam_ssa_private_append.erl, line 438)
  in call from maps:fold_1/4 (maps.erl, line 416)
  in call from beam_ssa_private_append:private_append/3 (beam_ssa_private_append.erl, line 72)
```

**Expected behavior**
Compile ok.

**Affected versions**
I tested with 26.1 and 26.2.5.1, I've also tested 26.0 and it's working fine.

**Additional context**
None.

--- a/lib/compiler/src/beam_ssa_private_append.erl
+++ b/lib/compiler/src/beam_ssa_private_append.erl
@@ -185,17 +185,10 @@ get_results([{Lbl,#b_blk{last=#b_ret{arg=#b_literal{val=Lit}}}}|Rest],
     %% appendable binary, it can happen that we encounter literals
     %% which do not match the type of the element. We can safely stop
     %% the tracking in that case.
-    Continue = case Element of
-                   {tuple_element,_,_} ->
-                       is_tuple(Lit);
-                   self ->
-                       is_bitstring(Lit);
-                   {hd,_} ->
-                       is_list(Lit) andalso (Lit =/= [])
-               end,
-    DefSt = if Continue ->
+    DefSt = case is_lit_compatible_with_element(Lit, Element) of
+                true ->
                     add_literal(Fun, {ret,Lbl,Element}, DefSt0);
-               true ->
+                false ->
                     DefSt0
             end,
     get_results(Rest, Acc, Element, Fun, DefSt);
@@ -203,6 +196,16 @@ get_results([_|Rest], Acc, Element, Fun, DefSt) ->
     get_results(Rest, Acc, Element, Fun, DefSt);
 get_results([], Acc, _, _Fun, DefSt) ->
     {Acc, DefSt}.
+
+is_lit_compatible_with_element(Lit, Element) ->
+    case Element of
+        {tuple_element,_,_} ->
+            is_tuple(Lit);
+        self ->
+            is_bitstring(Lit);
+        {hd,_} ->
+            is_list(Lit) andalso (Lit =/= [])
+    end.
 
 track_value_in_fun([{#b_var{}=V,Element}|Rest], Fun, Work,
                    Defs, ValuesInFun, DefSt0)
@@ -378,15 +381,17 @@ gca(Args, Element, Idx, Fun, Dst, DefSt) ->
 gca([#b_var{}=V|_], I, Element, I, _Fun, _Dst, DefSt) ->
     %% This is the argument we are tracking.
     {[{V,Element}], DefSt};
-gca([#b_literal{val=Lit}|_], I, self, I, _Fun, _Dst, DefSt)
-  when not is_bitstring(Lit)->
+gca([#b_literal{val=Lit}|_], I, Element, I, Fun, Dst, DefSt) ->
     %% As value tracking is done without type information, we can
     %% follow def chains which don't terminate in a bitstring. This is
     %% harmless, but we should ignore them and not, later on, try to
     %% patch them to a bs_writable_binary.
-    {[], DefSt};
-gca([#b_literal{val=Lit}|_], I, Element, I, Fun, Dst, DefSt) ->
-    {[], add_literal(Fun, {opargs,Dst,I+1,Lit,Element}, DefSt)};
+    case is_lit_compatible_with_element(Lit, Element) of
+        true ->
+            {[], add_literal(Fun, {opargs,Dst,I+1,Lit,Element}, DefSt)};
+        false ->
+            {[], DefSt}
+    end;
 gca([_|Args], I, Element, Idx, Fun, Dst, DefSt) ->
     gca(Args, I + 1, Element, Idx, Fun, Dst, DefSt).
 

--- a/lib/compiler/test/beam_ssa_check_SUITE_data/private_append.erl
+++ b/lib/compiler/test/beam_ssa_check_SUITE_data/private_append.erl
@@ -83,7 +83,8 @@
          bs_create_bin_on_literal/0,
 
          crash_in_value_tracking/3,
-         crash_in_value_tracking_inner/3]).
+         crash_in_value_tracking_inner/3,
+         gh8630/1]).
 
 %% Trivial smoke test
 transformable0(L) ->
@@ -1025,3 +1026,13 @@ crash_in_value_tracking(_, _V0, _) ->
     ((<<((crash_in_value_tracking_inner(
             {#{#{ ok => ok || _ := _ <- ok} => ok},
              _V0, false, _V0, "Bo"}, _V0, ok)))/bytes>>) =/= ok).
+
+gh8630(<<"\\",R/binary>>, Xs) ->
+%ssa% (_, _) when post_ssa_opt ->
+%ssa% _ = bs_init_writable(_).
+    gh8630(R, [<<>> | Xs]);
+gh8630(<<A:8,R/binary>>, [X|Xs]) ->
+    gh8630(R, [<<X/binary,A:8>> | Xs]).
+
+gh8630(I) ->
+    gh8630(I, []).

